sparkle/src/stories/Table.stories.tsx

```diff
@@ -0,0 +1,121 @@
+import type { Meta, StoryObj } from "@storybook/react";
+import React, { useCallback, useMemo,useState } from 'react';
+
+import {
+  FolderIcon,
+  Table,
+} from "../index_with_tw_base";
+
+const meta = {
+  title: "Components/Table",
+  component: Table
+} satisfies Meta<typeof Table>;
+
+
+type SortingState = {
+  column: DataKeys | null;
+  direction: "asc" | "desc";
+};
+
+type Data = {
+  name: string;
+  usedBy: number;
+  addedBy: string;
+  lastUpdated: string;
+  size: string;
+};
+
+type DataKeys = keyof Data;
+
+export default meta;
+type Story = StoryObj<typeof meta>;
+
+const TableExample = () => {
+  const initialData = [
+    {
+      name: "Marketing",
+      description: "(23 items)",
+      usedBy: 8,
+      addedBy: "User1",
+      lastUpdated: "July 8, 2023",
+      size: "32kb",
+      avatarUrl: "https://dust.tt/static/droidavatar/Droid_Lime_3.jpg",
+      clickable: true
+    },
+    { name: "Design", usedBy: 2, addedBy: "User2", lastUpdated: "2023-07-09", size: "64kb", icon: FolderIcon, clickable: true },
+    { name: "Development", usedBy: 5, addedBy: "User3", lastUpdated: "2023-07-07", size: "128kb", clickable: false },
+    { name: "Sales", usedBy: 10, addedBy: "User4", lastUpdated: "2023-07-10", size: "16kb", clickable: true },
+    { name: "HR", usedBy: 3, addedBy: "User5", lastUpdated: "2023-07-06", size: "48kb", clickable: false },
+  ];
+
+  const [sorting, setSorting] = useState<SortingState>({ column: null, direction: 'asc' });
+
+  const sortedData = useMemo(() => {
+    if (!sorting.column) {
+      return initialData;
+    }
+
+    return [...initialData].sort((a, b) => {
+      if (!sorting.column) {
+        return -1;
+      }
+      const aValue = a[sorting.column];
+      const bValue = b[sorting.column];
+
+      if (aValue < bValue) {
+        return sorting.direction === 'asc' ? -1 : 1;
+      }
+      if (aValue > bValue) {
+        return sorting.direction === 'asc' ? 1 : -1;
+      }
+      return 0;
+    });
+  }, [initialData, sorting]);
+
+
+  const handleSort = useCallback((column: DataKeys | null) => {
```

Comment:
The `DataTable` component should bundle all of this logic, sorting, filtering, ...